src/feed.tsx

```diff
@@ -8,18 +9,64 @@ export interface Properties {
   app: string;
 }
 
-export class Feed extends React.Component<Properties> {
+export interface State {
+  feed: FeedItemModel[];
+  hasMore?: boolean;
+  pageNumber?: number;
+  pageSize?: number;
+}
+
+export class Feed extends React.Component<Properties, State> {
+  componentDidMount() {
+    this.setState({ feed: this.props.items.slice(0, 2) });
+  }
+
+  componentDidUpdate(prevProps: Properties) {
+    const { items } = this.props;
+    if (items !== prevProps.items) {
+      this.setState({ feed: items.slice(0, 2) });
+      this.setState({ pageNumber: 2, pageSize: 2, hasMore: true });
+    }
+  }
+
+  fetchMoreData = () => {
+    const { pageNumber, pageSize, feed } = this.state;
+    if (feed.length >= this.props.items.length) {
+      this.setState({ hasMore: false });
+      return;
+    }
+    this.setState({
+      feed: feed.concat(
+        this.props.items.slice(
+          (pageNumber - 1) * pageSize,
```

Comment:
i don't think this will work like we're expecting for at least the initial render. try adding a test with more items than the page size, and i think it will show an unexpected number of items rendered, as well as which items are rendered.

(if we initialize page number to zero, we're going to get a negative number in that situation, which will slice from the end of the array. currently page number is not initialized, so i'm not even sure what that will result in, but i don't think it'll work how we want it to.)